core/controller/src/main/scala/org/apache/openwhisk/core/controller/Controller.scala

```diff
@@ -230,6 +233,10 @@ object Controller {
   def main(args: Array[String]): Unit = {
     implicit val actorSystem = ActorSystem("controller-actor-system")
     implicit val logger = new AkkaLogging(akka.event.Logging.getLogger(actorSystem, this))
+    if (useClusterBootstrap) {
```

Comment:
IIRC, the `FPCPoolBalancer` doesn't need any Akka-cluster feature.
Is there any reason to add this to the controller?

Reply:
The controller has an actor system now so if you replicate the controller, it needs to know how to discover different actors in the ```controller-actor-system```. Another reason I assumed this is needed is that ansible does configure a controller cluster (e.g., specified here: https://github.com/apache/openwhisk/blob/9bc5e043444d115cec0d79ac47d7002baa522924/ansible/controller.yml#L45). 

Before the new scheduler, the controller did not have an actor system at all (other than ```ShardingContaienrPoolBalancer``` which has similar logic to what I added).

To be fair, if you don't replicate the controller, this is not needed.

Reply:
AFAIK, `FPCPoolBalancer` does not need to discover different actors in a remote node.
The scheduler does. `ShardingPoolBalancer` needs a cluster to determine the number of controller nodes.
This is for sharding and `ShardingPoolBalancer` already has the same logic.

That's why I asked if akka-cluster bootstrap is required for all load balancers other than `ShardingPoolBalancer`.

Reply:
I had been concerned about multiple actors with the same name contacting the scheduler, but after further exploration, I do not think it was a valid concern.

I've removed the changes to the controller.